x/stake/handler.go

```diff
@@ -208,7 +171,7 @@ func handleMsgUnbond(ctx sdk.Context, msg MsgUnbond, k Keeper) sdk.Result {
 			return ErrNotEnoughBondShares(msg.Shares).Result()
 		}
 	} else {
-		if !bond.Shares.GT(shares) {
+		if bond.Shares.LT(shares) {
```

Comment:
Not the same when `bond.Shares == shares` - intentional?

Reply:
aha - this was something one of the tests caught - of course it's okay to unbond all of your shares, but `!bond.Shares.GT(shares)` catches a full unbond and reports it as an error.... the real error is only if there is less shares, not equal shares